Dynamic work section

Ayokunnumi1 opened the pull request:
These are the changes implemented in this branch;
* Dynamically created the work section of my html file,
* Added a pop-up-window,
* Made the pop-up-window responsive.

s2jack committed:
Declare function to load work section dynamically

Add loadworksection function
Add card01 elements into this function
Add function to run with data object that stores all data of projects when the page loads immediately

Ayokunnumi1 committed:
Fix linter errors

Ayokunnumi1 committed:
Fix eslint error

Ayokunnumi1 committed:
Create a pop-up-window

s2jack committed:
Add close and open pop up  function

Fix  style issues for mobile and desktop pop ups
Add functions for opening and closing modal window
Declare elements into main.js such as buttons and modal window etc
Add event listener for close icon and project buttons
Add images for project placeholders and close icon
Add new elements into index.html for desktop and mobile differences
  - new p  element for desktop description
  - new img element for desktop view
  - new ul element for desktop lists
Add dislay none properties for elements that belongs to only mobile or desktop
in the every media queries

Ayokunnumi1 committed:
Add image files & fix style errors

Ayokunnumi1 committed:
Merge branch 'main' into dynamic-work-section

Ayokunnumi1 committed:
Delete unnecessary index.html files & fix eslint errors

ridwanediallo commented on index.html:
```diff
@@ -259,6 +156,74 @@ <h2 class="contact--me">Contact me</h2>
+            printer took a galley of type and scrambled it 1960s.
+          </p>
+        </div>
+        <div class="pop-up-desc">
+          <p>
+            Lorem ipsum dolor sit amet consectetur adipisicing elit.
+            Error et delectus quae at alias voluptatum
+            debitis pariatur eveniet eius quam!
+          </p>
+        </div>
+        <div class="side-pop">
+          <ul class="skill-list-desktop">
+            <li>HTML</li>
+            <li>CSS</li>
+            <li>Javascript</li>
+            <li>Ruby</li>
+            <li>Bootstrap</li>
+          </ul>
+          <ul class="skill-list">
+            <li>HTML</li>
+            <li>CSS</li>
+            <li>Javascript</li>
+          </ul>
+          <div class="see-container">
+            <a href="">
+              <button>
+                <span>see live</span>
+                <img src="images/live-link-png.svg" alt="">
+              </button>
+            </a>
+            <a href="">
+              <button>
+                <span>see source</span>
+                <img style="width: 25px;" src="images/github-icon-png.png" alt="">
+              </button>
+            </a>
+          </div>
+        </div>
+      </div>
+  </section>
```
- [ ] Great job, team, on making your modal's design look similar to the template and making the open and close buttons work as expected 👏. However, kindly ensure that it is not hard coded in the HTML. The modal should be generated dynamically with Javascript. When the modal opens, it should be prefilled with the information (image, title, description, etc.) that was displayed on the card where we clicked the see project button. 👌
So as stated in the requirements, you need to have a  JavaScript array to store all of the information for all projects and use a loop to go through the array of objects and display the information of a card with the modal whenever a see project button is clicked.  👍️

Please review [the project requirements](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/8_popup.md) for this task, which I have attached a screenshot below. :point_down: 
#### Task requirement :point_down: 
![Screenshot from 2023-06-23 13-24-13](https://github.com/Ayokunnumi1/Portfolio-setup-and-mobile-first/assets/89228743/50968055-652b-41d4-abcf-962875ce2cb4)

ridwanediallo reviewed (changes requested):
# Changes Required ♻️
## Hi team, 
![AlmostThereReachGIF](https://user-images.githubusercontent.com/55918904/191301799-36d07166-0e35-4582-a231-7a2d780f7783.gif)

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Your modal design looks similar to the template. ✔
- Opening and closing the modal is woring correctly. ✔


## Required Changes ♻️

- [ ] Please update the project in the description section from **Hello Microverse** to the real name of the project to make your readme file look more professional. 🛠️🤓
#### Description section from readme file :point_down: 
![Screenshot from 2023-06-23 13-47-26](https://github.com/Ayokunnumi1/Portfolio-setup-and-mobile-first/assets/89228743/8f8bfb93-802c-443b-bcab-8dfa73872b53)



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

s2jack committed:
Add popUp elements into function to create dynamic

Declare opener and closer function for modal page into main.js
Delete all elements from index.html

s2jack committed:
Create popUp window dynamically

s2jack committed:
Try to fix errors

Ayokunnumi1 committed:
Create pop-up-window-dynamically & Update read-me file

Ayokunnumi1 committed:
Fix eslint errors

Ayokunnumi1 committed:
Merge branch 'dynamic-work-section' into modal-window-dynamic

Ayokunnumi1 committed:
Merge pull request #12 from Ayokunnumi1/modal-window-dynamic

Modal window dynamic

Ayokunnumi1 committed:
Create pop-up-window dynamically

Ayokunnumi1 committed:
Update Read-me file & delete unnecessary file

abdallahmalima commented on index.html:
```diff
@@ -253,7 +157,67 @@ <h2 class="contact--me">Contact me</h2>
     </form>
   </section>
   <section class="pop-up" id="modal">
-    
```
- [ ] 🙏Please update the code to dynamically populate the data for the pop-up display. Currently, it only displays the same data every time you click. This modification will allow users👨‍👩‍👧‍👦 to view the details of each specific project when they click the "See Project" button.

abdallahmalima reviewed (changes requested):
Hi @Ayokunnumi1 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

![A little bit more](https://user-images.githubusercontent.com/20554583/208236462-13ccd332-b921-416b-a8a9-b831ebd7c79b.gif)


 ## Highlights
- Popup appear✔️
- Popup close✔️
- Popup is responsive✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._✔️

Ayokunnumi1 committed:
Add descriptive commented lines in main.js
